### DISCUSSION TITLE
是否可以支持 Snowflake DB？

### DISCUSSION THREAD

**Discussion by xingwen1987** [dotnetcore/FreeSql#1570] [Ideas]
是否可以支持 Snowflake DB？

**Comment by 2881099** [comment]
```c#

var fsql = new FreeSqlBuilder()
    .UseConnectionFactory(DataType.CustomMySql, () => new SnowflakeDbConnection(...))
```


FreeSql.Provider.Custom 提供了这四种数据库适配，并且支持 CodeFirst/DbFirst 以及完整的 FreeSql 功能。

FreeSql.Provider.Custom 不依赖具体 ado.net/odbc/oledb dll 驱动，使用者在外部自行引用 dll 驱动。

访问 MySql 数据库为例：

```csharp
var fsql = new FreeSqlBuilder()
    .UseConnectionFactory(DataType.CustomMySql, () => 
        new MySqlConnection("Data Source=..."))
    .UseNoneParameter(true)
    .UseMonitorCommand(Console.WriteLine(cmd.CommandText))
    .Build();
fsql.SetDbProviderFactory(MySqlConnectorFactory.Instance);
```

若某国产数据库兼容 MySql SQL，先引用对方提供的 DLL，然后：

- 将上面 new MySqlConnection 替换成 new XxxConnection
- 将上面 MySqlConnectorFactory.Instance 替换成对应的 DbProviderFactory

提示：对方 DLL 一般都会提供这两个现实类
